internal/test/test_tpm.go

```diff
@@ -40,6 +41,26 @@ func (s simulatedTpm) EventLog() ([]byte, error) {
 	return s.eventLog, nil
 }
 
+// SkipOnAlgorithm skips the test if the algorithm is not found in the TPM
+// capability.
```

Comment:
If we think we'll want to use go-tpm's skip function to reduce duplication (which we would expose as a test package), we should add a TODO.

Reply:
Yes, we may also need to extend its functionality like supporting skipping over a ECC curve (CapabilityECCCurves). I will create an issue for that

Reply:
makes sense. please resolve when you do.